### ISSUE TITLE
Extra photos taken during turning in px4 

### ISSUE THREAD

**Issue by moreba1** [mavlink/qgroundcontrol#8430]
Hi

In turn around there is extra photos taken , the issue is in planing survey mission for px4 code .

![image](https://user-images.githubusercontent.com/14200369/75428794-fa09d380-595d-11ea-92cd-2935b4cb0c77.png)

**Comment by DonLakeFlyer** [comment]
Can you attach the plan file you are using?

**Comment by moreba1** [comment]
> 
> 
> Can you attach the plan file you are using?

yes , i attached plan . i think camera trigger distance must be separately after start waypoint not in camera section .


[plan.zip](https://github.com/mavlink/qgroundcontrol/files/4265500/plan.zip)

**Comment by DonLakeFlyer** [comment]
I looked at your plan and the mission generated has the correct camera start and stop command and the beginning and end of each transect. So this would be some sort of either firmware or camera problem. You'll need to talk with firmware folks.

**Comment by Antiheavy** [comment]
@moreba1 This is a know issue with PX4. @LorenzMeier and @DonLakeFlyer did a bunch of work on it a while back, but it seems to have been tricky to solve and maybe was dropped. https://github.com/PX4/Firmware/pull/11878

**Comment by LorenzMeier** [comment]
@DonLakeFlyer Have you implemented MAV_CMD_CONDITION_GATE consistently by now? If not, that part should still be open on the QGC side, as you otherwise have no clean definition of the transect.

**Comment by DonLakeFlyer** [comment]
> This is a known issue with PX4.

This is not related to MAV_CMD_CONDITION_GATE. The current problem with turning off images in turnaround is that the images stop too early not too late. This is because the vehicle hits the acceptance radius which triggers the state machine to move to the next step which is to stop triggering. This in turn can happen early enough that you miss photos at the edge of the survey area.

What is shown above is the opposite problem. The photos stop to late and in one case there is even a strange gap after transect exit. You can see from how the flight path begins to curve towards the turnaround that the mission state machine has moved past the survey edge waypoint which in turn means it also has processed the camera stop. Yet camera camera did not stop.

**Comment by moreba1** [comment]
> What is shown above is the opposite problem. The photos stop to late and in one case there is even a strange gap after transect exit. You can see from how the flight path begins to curve towards the turnaround that the mission state machine has moved past the survey edge waypoint which in turn means it also has processed the camera stop. Yet camera camera did not stop.

my problem is not  "images stop too early" , my problem is images start too early that cause we have images with 30 degree angles .

<img width="212" alt="a" src="https://user-images.githubusercontent.com/14200369/75631984-e20bab80-5c0c-11ea-8f06-9d2a9a36e6af.png">

**Comment by DonLakeFlyer** [comment]
Ooh, crap. Sorry not paying enough attention when I looked at your plan file. I thought the flight path was going the other direction. So this is the MAV_CMD_CONDITION_GATE implementation problem. Let me look at that again.

**Comment by moreba1** [comment]
does it done ?

**Comment by DonLakeFlyer** [comment]
Yes, but I'm not quite sure where the firmware support stands.

**Comment by moreba1** [comment]
> 
> 
> Yes, but I'm not quite sure where the firmware support stands.

@LorenzMeier
